internal/scheduler/scheduling/context/queue.go

```diff
@@ -58,8 +58,12 @@ type QueueSchedulingContext struct {
 	ScheduledResourcesByPriorityClass map[string]internaltypes.ResourceList
 	// Resources evicted from this queue during this scheduling cycle.
 	EvictedResourcesByPriorityClass map[string]internaltypes.ResourceList
+	// Resources preempted from this queue during this scheduling cycle.
+	PreemptedResourceByPriorityClass map[string]internaltypes.ResourceList
 	// Job scheduling contexts associated with successful scheduling attempts.
 	SuccessfulJobSchedulingContexts map[string]*JobSchedulingContext
+	// Job scheduling contexts associated with preempted jobs.
+	PreemptedJobSchedulingContexts map[string]*JobSchedulingContext
```

Comment:
Presumably this includes only jobs preempted by the optimiser, not those evicted but not re-scheduled? Suggest making that super clear from the variable name - `PreemptedResourceByPriorityClass ` to `PreemptedByOptimiserResourceByPriorityClass ` or similar.